assets/js/components/EntityHeader.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * EntityHeader component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { useWindowScroll } from 'react-use';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	useContext,
+	useCallback,
+	useEffect,
+	useRef,
+	useState,
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import ViewContextContext from './Root/ViewContextContext';
+import { VIEW_CONTEXT_PAGE_DASHBOARD } from '../googlesitekit/constants';
+import Button from './Button';
+import { CORE_SITE } from '../googlesitekit/datastore/site/constants';
+import BackspaceIcon from '../../svg/keyboard-backspace.svg';
+import { CORE_LOCATION } from '../googlesitekit/datastore/location/constants';
+import Link from './Link';
+const { useSelect, useDispatch } = Data;
+
+const EntityHeader = () => {
+	const viewContext = useContext( ViewContextContext );
+	const currentEntityTitle = useSelect( ( select ) =>
+		select( CORE_SITE ).getCurrentEntityTitle()
+	);
+	const entityURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getCurrentEntityURL()
+	);
+
+	const headerRef = useRef();
+	const [ url, setURL ] = useState( entityURL );
+
+	useEffect( () => {
+		const shortenURL = () => {
+			if ( ! headerRef.current ) {
+				return;
+			}
+
+			const entityHeaderWidth = headerRef.current.clientWidth;
+
+			const paddingLeft = global
+				.getComputedStyle( headerRef.current, null )
+				.getPropertyValue( 'padding-left' );
+			const paddingRight = global
+				.getComputedStyle( headerRef.current, null )
+				.getPropertyValue( 'padding-right' );
+			const entityHeaderPadding =
+				parseFloat( paddingLeft ) + parseFloat( paddingRight );
+
+			const headerDetailsWidth =
+				( entityHeaderWidth - entityHeaderPadding ) * 0.8; // headerDetails is 80% of header
+			const urlWidth = headerDetailsWidth - 20 - 16; // padding: entity-header__details = 20px, url <a> = 16px
+
+			const urlFontSize = global
+				.getComputedStyle( headerRef.current.lastChild.lastChild, null ) // url <a> element
+				.getPropertyValue( 'font-size' );
+			const size = parseFloat( urlFontSize );
+
+			// 2 is appox. the min character constant for sans-serif fonts:
+			// https://pearsonified.com/characters-per-line/
+			const maxChars = ( urlWidth * 2 ) / size;
```

Comment:
@jimmymadon why do we need all these calculations? Why not use the `clientWidth` property of the link to get the width of the element?

Reply:
@eugene-manuilov That's what we started with but as I mentioned [here](https://github.com/google/site-kit-wp/pull/4528#discussion_r769687358), the clientWidth, offsetWidth and getBoundingClientRect.width() were all not returning the correct width of the visible rendered <a> element. This is because React loads the element with the full entityURL first pretending to give it a longer width than what it has space for.

Reply:
Ah... ok, thanks for the clarification. Then, I think, if we add `max-width: 100%` property to the link styles, it will return the correct width even if the initial link is longer, right?